tutorials/Atmos/densitycurrent.jl

```diff
@@ -0,0 +1,367 @@
+    π_exner = FT(1) - _grav / (c_p * θ) * z             # exner pressure
+    ρ = p0 / (R_gas * θ) * (π_exner)^(c_v / R_gas)      # density
+    T = θ * π_exner
+    e_int = internal_energy(bl.param_set, T)
+    ts = PhaseDry(bl.param_set, e_int, ρ)
+    ρu = SVector(FT(0), FT(0), FT(0))                   # momentum
+    #State (prognostic) variable assignment
+    e_kin = FT(0)                                       # kinetic energy
+    e_pot = gravitational_potential(bl.orientation, aux)# potential energy
+    ρe_tot = ρ * total_energy(e_kin, e_pot, ts)         # total energy
+
+    # Assign State Variables
+    state.ρ = ρ
+    state.ρu = ρu
+    state.ρe = ρe_tot
+end
+
+# ## [Model Configuration](@id config-helper)
+# We define a configuration function to assist in prescribing the physical
+# model. The purpose of this is to populate the
+# [`ClimateMachine.AtmosLESConfiguration`](@ref LESConfig) with arguments
+# appropriate to the problem being considered.
+function config_densitycurrent(FT, N, resolution, xmax, ymax, zmax)
+
+    # Choose an Explicit Multi-rate Solver from the existing [ODESolvers](@ref
+    # ODESolvers-docs) options Apply the outer constructor to define the
+    # `ode_solver`. Here `AtmosAcousticGravityLinearModel` splits the
+    # acoustic-gravity wave components from the advection-diffusion dynamics.
+    # The 1D-IMEX method is less appropriate for the problem given the current
+    # mesh aspect ratio (1:1)
+    #=ode_solver = ClimateMachine.MultirateSolverType(
+        linear_model = AtmosAcousticGravityLinearModel,
+        slow_method = LSRK144NiegemannDiehlBusch,
+        fast_method = LSRK144NiegemannDiehlBusch,
+        timestep_ratio = 10,
+    )=#
+    ode_solver = ClimateMachine.ExplicitSolverType(
+        solver_method = LSRK144NiegemannDiehlBusch,
+    )
+
+    # Since we want four tracers, we specify this and include the appropriate
+    # diffusivity scaling coefficients (normally these would be physically
+    # informed but for this demonstration we use integers corresponding to the
+
+    # The model coefficient for the turbulence closure is defined via the
+    # [CLIMAParameters
+    # package](https://CliMA.github.io/CLIMAParameters.jl/latest/) A reference
+    # state for the linearisation step is also defined.
+    T_surface = FT(300)
+    T_min_ref = FT(0)
+    T_profile = DryAdiabaticProfile{FT}(param_set, T_surface, T_min_ref)
+    ref_state = HydrostaticState(T_profile)
+
+    # The fun part! Here we assemble the `AtmosModel`.
+    #md # !!! note
+    #md #     Docs on model subcomponent options can be found here:
+    #md #     - [`param_set`](https://CliMA.github.io/CLIMAParameters.jl/latest/)
+    #md #     - [`turbulence`](@ref Turbulence-Closures-docs)
+    #md #     - [`hyperdiffusion`](@ref Hyperdiffusion-docs)
+    #md #     - [`source`](@ref atmos-sources)
+    #md #     - [`tracers`](@ref Tracers-docs)
+    #md #     - [`init_state`](@ref init)
+
+    _C_smag = FT(0.21)
```

Comment:
```suggestion
    _C_smag = FT(C_smag(param_set))
```

Reply:
> 

As of now _C_smag = FT(C_smag(param_set)) breaks the run whereas _C_smag = FT(0.21) doesn't. It needs to be checked.

Reply:
It shouldn't, CLIMAParameters [uses the same value](https://github.com/CliMA/CLIMAParameters.jl/blob/master/src/AtmosParameters.jl#L4). The other option is to [override the default](https://clima.github.io/CLIMAParameters.jl/dev/#Overriding-defaults-1), but this shouldn't even be needed.